runtime/src/accounts_db.rs

```diff
@@ -1773,8 +1941,21 @@ impl AccountsDB {
         let mut total_removed_bytes = 0;
 
         let mut remove_storages_elapsed = Measure::start("remove_storages_elapsed");
-        for slot in non_roots {
-            if let Some((_, slot_removed_storages)) = self.storage.0.remove(&slot) {
+
+        // Remove the backing storages for the dead slots. The accounts index cleaning (removing
+        // from the slot list, decrementing the account ref count), is handled in
+        // clean_accounts() -> purge_older_root_entries()
```

Comment:
let's move this comment into the `Some((_, slot_removed_storages)) = self.storage.0.remove(&remove_slot)` clause below?

That's because the accounts index cleaning is actually done for the slot_cache by `clean_dead_slots_common` via `purge_slot_cache_keys`.